tests/Unit/Routing/PagedRouteCollectionTest.php

```diff
@@ -47,12 +47,18 @@ public function testIterator($amountRoutes, $routesLoadedInParallel, $expectedCa
         }
         $names = array_keys($routes);
 
+        $counter = 0;
+        $with = [];
+        $will = [];
         foreach ($expectedCalls as $i => $range) {
-            $this->routeProvider->expects($this->at($i))
-              ->method('getRoutesPaged')
-              ->with($range[0], $range[1])
-              ->will($this->returnValue(array_slice($routes, $range[0], $range[1])));
+            $with[] = [$range[0], $range[1]];
+            $will[] = array_slice($routes, $range[0], $range[1]);
+            ++$counter;
         }
+        $mocker = $this->routeProvider->expects($this->exactly($counter))
```

Comment:
we could also `\count($expectedCalls)` instead of having a separate counter. i would find that a bit more straightforward to read, if you don't mind changing it.

Reply:
both fixed, thanks for review